Cython/Compiler/Nodes.py

```diff
@@ -2869,18 +2884,26 @@ def is_cdef_func_compatible(self):
 
     def analyse_declarations(self, env):
         if self.decorators:
-            for decorator in self.decorators:
+            # should only apply to functions outside cdef classes
+            if len(self.decorators):
+                # only test the innermost decorator for classmethod
+                # and static method - anything else will have to be
+                # fed to the relevant builtin functions
+                decorator = self.decorators[-1]
                 func = decorator.decorator
                 if func.is_name:
                     self.is_classmethod |= func.name == 'classmethod'
                     self.is_staticmethod |= func.name == 'staticmethod'
-
-        if self.is_classmethod and env.lookup_here('classmethod'):
+        cm_entry = env.lookup('classmethod')
```

Comment:
Why is `lookup()` more correct than `lookup_here()` in this case?

Reply:
`lookup_here` only checks inside the class scope so it's almost never right. It won't notice if you've re-defined `classmethod` at module level, which has to be much more common. e.g.

```
classmethod = lambda x: (lambda: 1)

cdef class C:
    @classmethod
    def cm(cls):
        pass
```